File: AdminService/src/main/java/com/ecommercial/shopping/adminservice/global/controller/BaseController.java
package com.ecommercial.shopping.adminservice.global.controller;

import com.ecommercial.shopping.adminservice.global.dto.BaseResponse;
import com.ecommercial.shopping.adminservice.global.exception.MyException;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class BaseController {

    @ExceptionHandler(MyException.class)
    public ResponseEntity exceptionHandler(MyException e) {
        return ResponseEntity.status(e.getExceptionCode()).body(new BaseResponse("error", e.getMessage()));
    }
}

File: OrderService/src/main/kotlin/com/ecommercial/shopping/orderservice/global/controller/BaseController.kt
package com.ecommercial.shopping.orderservice.global.controller

import com.ecommercial.shopping.orderservice.global.dto.BaseResponse
import com.ecommercial.shopping.orderservice.global.exception.ErrorException
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class BaseController {

    @ExceptionHandler(ErrorException::class)
    fun exceptionHandler(e: ErrorException): ResponseEntity<BaseResponse<String>> {
        val response = BaseResponse<String>("error", e.errorMessage)
        return ResponseEntity(response, e.httpStatus)
    }
}